### PR TITLE
RESTWS-895: Return 204 response when the patient allergy status is unknown

### PR DESCRIPTION
**Pull request by kdaud**
## Issue I worked on

see https://issues.openmrs.org/browse/RESTWS-895

## Checklist: I completed these to help reviewers :)

- [x] My pull request only contains **ONE single commit**
(the number above, next to the 'Commits' tab is 1).

  No? -> [read here](https://wiki.openmrs.org/display/docs/Pull+Request+Tips) on how to squash multiple commits into one

- [x] My IDE is configured to follow the [**code style**](https://wiki.openmrs.org/display/docs/Java+Conventions) of this project.

  No? Unsure? -> [configure your IDE](https://wiki.openmrs.org/display/docs/How-To+Setup+And+Use+Your+IDE), format the code and add the changes with `git add . && git commit --amend`

- [ ] I have **added tests** to cover my changes. (If you refactored
  existing code that was well tested you do not have to add tests)

  No? -> write tests and add them to this commit `git add . && git commit --amend`

- [x] I ran `mvn clean package` right before creating this pull request and
  added all formatting changes to my commit.

  No? -> execute above command

- [ ] All new and existing **tests passed**.

  No? -> figure out why and add the fix to your commit. It is your responsibility to make sure your code works.

- [x] My pull request is **based on the latest changes** of the master branch.

  No? Unsure? -> execute command `git pull --rebase upstream master`

### CODE REVIEW

File: omod-common/src/main/java/org/openmrs/module/webservices/rest/web/response/ObjectNotFoundException.java [hunk 1]
```diff
@@ -9,6 +9,8 @@
  */
 package org.openmrs.module.webservices.rest.web.response;
 
+import java.util.Date;
```
Comment (dkayiwa) [inline omod-common/src/main/java/org/openmrs/module/webservices/rest/web/response/ObjectNotFoundException.java:12]: What are the changes in this file for?

File: omod-common/src/main/java/org/openmrs/module/webservices/rest/web/response/ResponseStatusException.java [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+ *
+ * Copyright (C) OpenMRS Inc. OpenMRS is a registered trademark and the OpenMRS
+ * graphic logo is a trademark of OpenMRS Inc.
+ */
+package org.openmrs.module.webservices.rest.web.response;
+
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Controller;
+import org.springframework.web.bind.annotation.ExceptionHandler;
```
Comment (dkayiwa) [inline omod-common/src/main/java/org/openmrs/module/webservices/rest/web/response/ResponseStatusException.java:15]: What is the new class for?

Comment (kdaud) [reply omod-common/src/main/java/org/openmrs/module/webservices/rest/web/response/ResponseStatusException.java:15]: I want to create a method with a custom error message that I can call out to do the work.

File: omod-common/src/main/java/org/openmrs/module/webservices/rest/web/response/ObjectNotFoundException.java (hunk 1 above)
Comment (kdaud) [reply omod-common/src/main/java/org/openmrs/module/webservices/rest/web/response/ObjectNotFoundException.java:12]: I want to rather use a customized constructor.

File: omod-common/src/main/java/org/openmrs/module/webservices/rest/web/response/ObjectNotFoundException.java [hunk 3]
```diff
@@ -20,12 +20,13 @@
 public class ObjectNotFoundException extends ResponseException {
 	
 	private static final long serialVersionUID = 1L;
-
```
Comment (dkayiwa) [inline omod-common/src/main/java/org/openmrs/module/webservices/rest/web/response/ObjectNotFoundException.java:23]: Can you discard the changes in this file?

**Comment by dkayiwa** [comment]
With your changes, what happens when one requests for allergies for a patient whose status is unknown?

**Comment by kdaud** [comment]
Nothing is being return to the client.

**Comment by dkayiwa** [comment]
> Nothing is being return to the client.

Then how how are you helping the reporter of the problem on talk?

**Comment by kdaud** [comment]
@dkayiwa based on the responses on talk, this commit is now ok to go into master after testing it out.

**Comment by dkayiwa** [comment]
Did you see Burke's response?

**Comment by kdaud** [comment]
Yes!

**Comment by dkayiwa** [comment]
Did you also include the tests that i told you?

**Comment by dkayiwa** [comment]
@kdaud is there a reason why you make a million commits on pull requests like this? Am not against the concept of commit early and often. But your rate of committing on such pull requests looks like you do not first test locally before committing.

**Comment by kdaud** [comment]
I am sorry! Now onward will be testing locally before pushing up.

**Comment by kdaud** [comment]
@dkayiwa have added the test however, its failing as a result of failing to locate the resource. Am using `allergy` as a resource path though this is failing, what could be the right resource path to be passed in the get() method?

**Comment by dkayiwa** [comment]
@kdaud your test should not be in that class. It should instead be in the `PatientAllergyController2_0Test` class.

**Comment by dkayiwa** [comment]
@kdaud when are you gonna finish this?

**Comment by kdaud** [comment]
today!

**Comment by kdaud** [comment]
Closing this PR in favor of pull/561

**Pull request closed without merging**